android/app/src/main/java/io/mosip/registration_client/api_services/AuthenticationApi.java

```diff
@@ -0,0 +1,206 @@
+                        this.context.getString(R.string.app_name),
+                        Context.MODE_PRIVATE);
+    }
+
+    private void doLogin(final String username, final String password, AuthResponsePigeon.Result<AuthResponsePigeon.AuthResponse> result) {
+        Call<ResponseWrapper<String>> call = syncRestService.login(syncRestFactory.getAuthRequest(username, password));
+        call.enqueue(new Callback<ResponseWrapper<String>>() {
+            @Override
+            public void onResponse(Call call, Response response) {
+                ResponseWrapper<String> wrapper = (ResponseWrapper<String>) response.body();
+                if (response.isSuccessful()) {
+                    ServiceError error = SyncRestUtil.getServiceError(wrapper);
+                    if (error == null) {
+                        try {
+                            loginService.saveAuthToken(wrapper.getResponse(), username);
+                            loginService.setPasswordHash(username, password);
+                            AuthResponsePigeon.AuthResponse authResponse = new AuthResponsePigeon.AuthResponse.Builder()
+                                    .setResponse(wrapper.getResponse())
+                                    .setUsername(sharedPreferences.getString(USER_NAME, null))
+                                    .setIsDefault(sharedPreferences.getBoolean(IS_DEFAULT, false))
+                                    .setIsOfficer(sharedPreferences.getBoolean(IS_OPERATOR, false))
+                                    .setIsSupervisor(sharedPreferences.getBoolean(IS_SUPERVISOR, false))
+                                    .build();
+                            result.success(authResponse);
+                            return;
+                        } catch (InvalidMachineSpecIDException e) {
+                            error = new ServiceError("", "Invalid Machine Spec ID found");
+                            Log.e(getClass().getSimpleName(), "Failed to save auth token", e);
+                        } catch (Exception e) {
+                            error = new ServiceError("", e.getMessage());
+                            Log.e(getClass().getSimpleName(), "Failed to save auth token", e);
+                        }
+                    }
+
+                    String errorCode = "";
+                    if (error == null) {
+                        errorCode = "REG_TRY_AGAIN";
+                    } else if (error.getMessage().equals("Invalid Request")) {
+                        errorCode = "REG_INVALID_REQUEST";
+                    } else if (error.getMessage().equals("Machine not found")) {
+                        errorCode = "REG_MACHINE_NOT_FOUND";
+                    } else {
+                        errorCode = error.getMessage();
+                    }
+                    AuthResponsePigeon.AuthResponse authResponse = new AuthResponsePigeon.AuthResponse.Builder()
+                            .setResponse("")
```

Comment:
the same error response is duplicated more than once in this class. Can we write a private method in this class, getAuthErrorResponse(String errorCode), and use that in all the error and exception blocks?

Reply:
Added a private method getAuthErrorResponse(String errorCode)